src/lib/util/CategoryUtil.php

```diff
@@ -492,12 +492,13 @@ public static function deleteCategoryByID($cid)
     {
         $em = \ServiceUtil::get('doctrine.entitymanager');
 
-        $dql = "DELETE FROM Zikula\CategoriesModule\Entity\CategoryEntity c WHERE c.id = :cid";
+        // delete attributes first to avoid error about foreign key constraints
+        $dql = "DELETE FROM Zikula\CategoriesModule\Entity\CategoryAttributeEntity a WHERE a.category = :cid";
         $query = $em->createQuery($dql);
         $query->setParameter('cid', $cid);
         $query->getResult();
 
-        $dql = "DELETE FROM Zikula\CategoriesModule\Entity\CategoryAttributeEntity a WHERE a.category = :cid";
+        $dql = "DELETE FROM Zikula\CategoriesModule\Entity\CategoryEntity c WHERE c.id = :cid";
```

Comment:
If entity is properly set up, manual DQL would not be required but simply cascade when using `->remove()`